### ISSUE TITLE
Cruiser and/or gunboat lag

### ISSUE THREAD

**Issue by AssjackGames** [OpenRA/OpenRA#18475]
## Issue Summary

DESCRIPTION

Not sure if this has been brought up elsewhere, but from my cursory search I hadn't seen someone point this out.

So everyone knows the game will lag (multiplayer, big matches) causing everything to come to a crawl for everyone connected, but I and my friends noticed this seems to be caused specifically by many many cruisers and/or gunboats sitting in bays. When the AI makes a massive swarm of cruisers (I think this boat specifically is the issue) they always build an insane number (like 100), and causes server-wide lag. When I nuke their bay and rid the map of cruisers the game speed returns.

## Reproduction

STEPS TO REPRODUCE THE ISSUE

Red Alert (original), multiplayer, big matches (7+ players). Seems consistently reproducible when the above scenario happens. Never tried making myself a swarm of cruisers that big so I would think it's the unit, not the general AI. I do build massive swarms with tanks, infantry, etc, just seems to be the cruiser and/or gunboat.

## Expected behavior

EXPECTED BEHAVIOR

Not to lag.

**Comment by Mailaender** [comment]
Which map were you playing?

**Comment by pchote** [comment]
This is a known issue, IIRC a regression of the AttackTurreted rework. I had a PR a few months ago (but no time to try and find it) that partially mitigated this and had an explanation of why this can't easily be fixed further.

**Comment by AssjackGames** [comment]
> Which map were you playing?

I've experienced this on a few maps. All extra large ones, like the Europe maps on the maps community page.


> This is a known issue, IIRC a regression of the AttackTurreted rework. I had a PR a few months ago (but no time to try and find it) that partially mitigated this and had an explanation of why this can't easily be fixed further.

Good to hear it's at least known about and being worked on.

**Comment by RoosterDragon** [comment]
We've made some improvements to the AI and pathfinding. Previously the AI was trying to order crusiers to reach places they couldn't get to, and the continuous pathfinding attempts were causing lag. We've fixed the AI so it no longer tries to order units to places they can't reach (#20227), and improved the pathfinder to reduce the performance impact. The fixes will be available in the next release.

I ran an AI game on one of the Europe maps for 10 mins on commit https://github.com/OpenRA/OpenRA/commit/5157bc375d554e275b142e650fec27356a8bd63b, which has all the fixes applied. [ra-2023-11-04T105726Z.zip](https://github.com/OpenRA/OpenRA/files/13256599/ra-2023-11-04T105726Z.zip)

No AI units got stuck in place, and no lag spikes were seen. The game ran smoothly. We can close this issue.
